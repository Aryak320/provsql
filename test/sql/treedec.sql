\set ECHO none
SET search_path TO provsql_test,provsql;

CREATE TABLE treedec_result AS
SELECT city, probability_evaluate(provenance(),'tree-decomposition') AS prob
FROM (
  SELECT DISTINCT city
  FROM personnel
EXCEPT
  SELECT p1.city
  FROM personnel p1,personnel p2
  WHERE p1.id<p2.id AND p1.city=p2.city
  GROUP BY p1.city
) t
ORDER BY CITY;

SELECT remove_provenance('treedec_result');

SELECT city, ROUND(prob::numeric,2) AS prob FROM treedec_result;
DROP TABLE treedec_result;

CREATE TABLE treedec_result AS
  SELECT name, city, shapley(c.provenance,p.provenance,'tree-decomposition') FROM (
    SELECT provenance() from (SELECT DISTINCT 1 FROM (
      (SELECT DISTINCT city FROM personnel)
    EXCEPT
      (SELECT p1.city
      FROM personnel p1, personnel p2
      WHERE p1.city = p2.city AND p1.id < p2.id
      GROUP BY p1.city
      ORDER BY p1.city)
      ) t
    ) u)
  AS c,
  (SELECT *, provenance() FROM personnel) AS p;

SELECT remove_provenance('treedec_result');

SELECT name, city, ROUND(shapley::numeric,3) AS shapley FROM treedec_result
ORDER BY city, name;

DROP TABLE treedec_result;
