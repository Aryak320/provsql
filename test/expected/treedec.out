\set ECHO none
 remove_provenance 
-------------------
 
(1 row)

   city   | prob 
----------+------
 Berlin   | 0.54
 New York | 0.26
 Paris    | 0.41
(3 rows)

 remove_provenance 
-------------------
 
(1 row)

   name   |   city   | shapley 
----------+----------+---------
 Ellen    | Berlin   |   0.093
 Susan    | Berlin   |   0.264
 John     | New York |   0.037
 Paul     | New York |   0.080
 Dave     | Paris    |   0.056
 Magdalen | Paris    |   0.113
 Nancy    | Paris    |   0.156
(7 rows)

